x/gov/keeper/proposal.go

```diff
@@ -43,11 +52,22 @@ func (keeper Keeper) SubmitProposal(ctx context.Context, messages []sdk.Msg, met
 	for _, msg := range messages {
 		msgsStr += fmt.Sprintf(",%s", sdk.MsgTypeURL(msg))
 
-		// TODO HV2: sdk.Msg has thousands of implementations that theoretically can pass the proposal
-		//  out of those, only some hundreds have `HasValidateBasic` function that could fail the proposal at `ValidateBasic`
-		//  We need to exclude here all the sdk.Msg implementations that we won't accept in gov module.
-		//  Search for RegisterImplementations((*sdk.Msg) for all modules and only include the ones that are relevant for heimdall v2
-		//  Is it only ParamsUpdate?
+		// TODO HV2: double check this
+		// HV2: filter out the message types that are not supported by the gov module
+		// only accept `MsgUpdateParams` for the heimdall-v2 enabled modules
+		switch msg.(type) {
+		case *v1.MsgUpdateParams, *govv1.MsgUpdateParams,
```

Comment:
I would probably declare these somewhere like in a const file in `/types`

Reply:
Good one, will do.

Reply:
[Done](39fd8a993160d716f503d10c9f5567f397c4605f), but in a file inside `keeper` package because in types this was creating a circular dependency.